capstone/src/main/java/com/google/univiz/CollegeDataApi.java

```diff
@@ -0,0 +1,9 @@
+package com.google.univiz;
+
+import java.util.List;
+import java.util.Set;
+
+public interface CollegeDataApi {
+
+  List<CollegeData> getCollegesById(Set<String> ids, Set<String> names);
```

Comment:
Why are `names` required? How will they be used?

Reply:
Back when i was thinking about this method, i assumed it would get the list of field names that it should return. But i believe we later decided that it should return all fields and later resource classes would down-sample it to the desired autovalue. So the second argument was the desired field names. For example MapResouce class would call this method and only ask for lat, lng, and name fields to be returned. You can remove it for now and if we wanted to take this route (to improve performance) we can add it back later.